### PR TITLE
Rework JNI CMake to leverage rapids_find_package

### PR DESCRIPTION
**Pull request by jlowe**
The JNI CMakeLists.txt has been fragile, looking for specific .a or .so libraries and header file locations, and will break again when libcudf moves to a pre-installed nvcomp 2.3 package since it expects to find nvcomp in a very specific location today.  This refactors the JNI CMakeLists.txt to leverage `rapids_find_package` to reuse the work performed in the libcudf build and also has the nice side-effect of avoiding redundant pulls and builds of the Thrust and RMM repositories that is happening today.

Another side-effect is that the JNI will now automatically pull in the same RMM compile definitions that are used for libcudf, meaning the separate RMM logging flag for the JNI build is no longer necessary.  Similarly it's no longer necessary to explicitly specify to the JNI build which type of Arrow library to use (i.e,.: static or dynamic), it will automatically use whichever Arrow library was built by libcudf.

### CODE REVIEW

**Review by revans2 — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

File: java/ci/build-in-docker.sh [hunk 1]
```diff
@@ -78,7 +78,6 @@ BUILD_ARG="-Dmaven.repo.local=\"$WORKSPACE/.m2\"\
  -DPER_THREAD_DEFAULT_STREAM=$ENABLE_PTDS\
  -DCUDA_STATIC_RUNTIME=$ENABLE_CUDA_STATIC_RUNTIME\
  -DCUDF_JNI_LIBCUDF_STATIC=ON\
- -DRMM_LOGGING_LEVEL=$RMM_LOGGING_LEVEL\
```
Comment (gerashegalov) [inline java/ci/build-in-docker.sh:81]: Do we need to make similar changes in [build.sh](https://github.com/rapidsai/cudf/blob/branch-22.06/build.sh#L151)?

**Review by revans2 — APPROVED** [review]

**Comment by jlowe** [comment]
Note that the previous CI failures were due to problems with the cuco dependency in a shared library build setup which should be addressed by #10662.

File: java/ci/build-in-docker.sh (hunk 1 above)
Comment (jlowe) [reply java/ci/build-in-docker.sh:81]: Nice catch!  This should now be fixed.

**Review by gerashegalov — APPROVED** [review]

**Comment by jlowe** [comment]
@gpucibot merge

**Pull request merged**